Sources/Amplitude/Amplitude.m

```diff
@@ -1073,6 +1091,10 @@ - (void)enterBackground {
         self->_inForeground = NO;
         [self refreshSessionTime:now];
         [self uploadEventsWithLimit:0];
+        [AMPStorage finish:[AMPStorage getDefaultEventsFile]];
```

Comment:
Need to verify if this will be triggered when killing the app.

Reply:
If I have no events at all, will this crash my app?

Reply:
This `finish` method only does work if the appropriate file exists.

Reply:
What would happen if this not get triggered? Like killed by memory management? That time this may not be triggered.

Reply:
That's a good catch. One simple way of handling that is to run `finish` if we see a JSON parsing error anywhere, and then try again